### PR TITLE
Bad pixel correction

### LINKED ISSUE #10: Bad pixel correction
Add a string choice parameter with options “masking” (set to 0), “replacement” (use the model values) and in the future “interpolation” (in case no model value is available). The default should be "replacement".

### PR DESCRIPTION
**Pull request by AntoineDarveau**
Fixes #10 Add possibility to correct bad pixel with the trace models. This option is the default.

### CODE REVIEW

**Comment by loicalbert** [comment]
Merci Antoine. The logic looks good to me. I haven't tested the code and won't be for the near future.

**Comment by loicalbert** [comment]
OK with the request.

**Review by loicalbert — APPROVED** [review]
I have read the code. Many changes were made. The logic is good. I am not in a position to run the code to test it.

**Comment by AntoineDarveau** [comment]
Question:  Should we include the parameter to deal with bad pixels ('model' or 'masking') in Extract1dStep or just leave it with the default being 'model' when models are available?

**Comment by loicalbert** [comment]
I think it should be up one level such that we can modify the argument from the DMS step Extract1dStep. That way, users could alter it. It otherwise would not be possible for them to do so.

**Review by talensgj — APPROVED** [review]
I've not read the code in detail, but it seems to run fine.

I did see 2 small things, but I don't think either of them are essential. Consider them pointers for future code.

`order_list = (1, 2, 3)` that's a tuple, I personally would have used a list, though I don't think it matters in this case. It might matter if this becomes a parameter per the TO DO item.

`decont[scimask] = tracemodels[order][scimask]` for an action like this I would recommend using `decont = np.where(scimask, tracemodels[order], decont)`

File: jwst/extract_1d/soss_extract/soss_extract.py [hunk 1]
```diff
@@ -259,60 +259,88 @@ def model_image(scidata_bkg, scierr, scimask, refmask, ref_files, transform=None
...
+        'model' option uses `tracemodels` to replace the bad pixels.
+    Returns
+    -------
+    wavelengths, fluxes, fluxerrs, npixels
+    Each is a dictionary with each extracted orders as key.
     """
+    # TODO Make it an input? So it can change for substrip96
+    # Which orders to extract.
+    order_list = (1, 2, 3)
+    order_str = {ord: f'Order {ord}' for ord in order_list}
```
Comment (njcuk9999) [inline jwst/extract_1d/soss_extract/soss_extract.py]: Note `ord` is a python reserved keyword - I wouldn't overwrite it

**Comment by AntoineDarveau** [comment]
Question: How to deal with the error for the masked pixels? 
Options I see: 
1. Put it to zero, so no effect on the uncertainties?
2. Estimate based on the fit on the other non masked pixels, on the same column (so they cover a similar range).
3. Use other valid pixel to estimate the uncertainty empirically

**Pull request merged**